### PR TITLE
Automate contributors & assist changelog

### LINKED ISSUE #238: ANSI-escaped outputs do not show in color
I just tried Script and found that ANSI-escaped outputs do not show in color. 
I don't know JS, but it seems that PR #218 breaks the functionality: 
in line 266 of `lib/script-view.coffee`

```
if atom.config.get('script.escapeConsoleOutput')
```

`script.escapeConsoleOutput` is not defined/accessible.

### PR DESCRIPTION
**Pull request by rgbkrk**
Initial stabs at #255 and #254.

This modifies package.json in place before publishing. We may want to just use this as a manual command.

This outputs some information to summarize the CHANGELOG, using the messages from the commits that came after the most recent release:

```console
$ node utils/updatePackage.js
Summarize these for the CHANGELOG:
>>>>>>>>>>>>>>>
Newline at the end.

Update package.json, using name:, email: format

Summarize the CHANGELOG.

Clean up intermediary work.

Write out contributors to package.json

Put the update package in as a prepublish step.

Grab the most recent email to be used, rank devs

Initial stabbing at #254

Create CONTRIBUTING.md
Merge pull request #242 from erran/ansi-escape-modifications

Fix #238 - Convert ANSI codes even with escapeConsoleOutput unset
Fix #238 - Convert ANSI codes even with escapeConsoleOutput unset

* Update configuration defaults
* Revert the conditional escape logic to exclude the ANSI toHTML
statement (thanks for pointing this out @morinmorin)

<<<<<<<<<<<<<<<
package.json written
```

What do you think @erran? This requires nodegit.

### CODE REVIEW

File: utils/updatePackage.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+var open = require("nodegit").Repository.open;
```
Comment (erran) [inline utils/updatePackage.js]: I have to ask: "why JavaScript?"

**Comment by erran** [comment]
> What do you think @erran? This requires nodegit.

Sure does, I had to install it globally after having the local `npm install`. The prepublish task runs when using `npm install` which was a little surprising to me.

File: package.json [hunk 2]
```diff
@@ -83,5 +202,8 @@
   },
   "devDependencies": {
     "grunt": "~0.4.5"
+  },
+  "scripts": {
+    "prepublish": "node utils/updatePackage.js"
```
Comment (erran) [inline package.json]: Let's change this to a different phase since "prepublish" runs on `npm install` as well.

File: package.json [hunk 3]
```diff
@@ -83,5 +202,8 @@
   },
   "devDependencies": {
     "grunt": "~0.4.5"
```
Comment (erran) [inline package.json]: Let's add "nodegit" as a dev dependency.

File: package.json (hunk 2 above)
Comment (erran) [inline package.json]: We could create a "prepare-release" phase or similar.

**Comment by erran** [comment]
See the script I wrote to generate the CHANGELOG.md section: [changes-between.sh](https://gist.github.com/erran/73bee2404c725c4b1cf5)

```markdown
## 2.16.0
* Update CHANGELOG.md
* Rust.
* Create CONTRIBUTING.md
* Add an anchor to the D grammar in README.md
* added D information to readme
* added file based grammar for D
* Add examples/colored_diagnostics.c
* Turn on colored diagnostics for the C language family
* Clear process on kill, don't use BufferedProcess.process
* Set bufferedProcess to null on process exit or run error
* Fix #238 - Convert ANSI codes even with escapeConsoleOutput unset
* Update CHANGELOG.md
* Fix #232 - Remove unused less import "ui-mixins"
* Add anchors to the limitations section
```

File: package.json (hunk 2 above)
Comment (rgbkrk) [inline package.json]: Ah, ok. I didn't know about that phase. I'm also a fan of removing this for now and making it manual.

File: package.json (hunk 3 above)
Comment (rgbkrk) [inline package.json]: Ah, right.

**Comment by rgbkrk** [comment]
Won't `git log v2.14.0..v2.15.0` just give you the changes between 2.14 and 2.15? You probably just need a changes_since script that just goes back to the last one.

```
git log $1.. --pretty="format:* %s" | egrep -v '(Merge pull request|Prepare \d+\.\d+\.\d+ release)'
```

Unless you're making the CHANGELOG after the release.

Do you want me to tear the changelog bits out of this one or just do the same as your `egrep -v` does in knocking junk out?

**Comment by erran** [comment]
> Won't git log v2.14.0..v2.15.0 just give you the changes between 2.14 and 2.15? You probably just need a changes_since script that just goes back to the last one.

You're right.

> Do you want me to tear the changelog bits out of this one or just do the same as your egrep -v does in knocking junk out?
> 
I think we should use egrep -v to knock out as much junk as possible but we should still revise the changelog before committing as some messages are less helpful than others. We may also want to start tagging messages with the PR where they were introduced.

File: utils/updatePackage.js (hunk 1 above)
Comment (rgbkrk) [inline utils/updatePackage.js]: How did I ever miss this question? :tongue:

I'm... not sure, other than in knowing that it was likely available to Atom users, I'm not sure. Perhaps I should have written coffeescript.

**Comment by rgbkrk** [comment]
Well @erran, I've finally updated this with some changes. We should set up a `.mailmap` file to deal with the duplicate emails. That doesn't work with nodegit though.

**Pull request merged**